### ISSUE TITLE
Incorrect code generated for data classes with positional parameters

### ISSUE THREAD

**Issue by ykrasik** [BreX900/data_class#7]
Hi, thanks for this package!

When trying to use it I noticed that it doesn't support classes with positional parameters and expects all parameters to be named.

```dart
@DataClass()
class Test with _$Test {
  final int positional1;
  final String positional2;
  final bool named1;

  const Test(this.positional1, this.positional2, {this.named1 = true});
}
```

generates 

```dart
mixin _$Test {
  Test get _self => this as Test;

  Iterable<Object?> get _props sync* {
    yield _self.positional1;
    yield _self.positional2;
    yield _self.named1;
  }

  bool operator ==(Object other) =>
      identical(this, other) ||
      other is Test &&
          runtimeType == other.runtimeType &&
          DataClass.$equals(_props, other._props);

  int get hashCode => Object.hashAll(_props);

  String toString() => (ClassToString('Test')
        ..add('positional1', _self.positional1)
        ..add('positional2', _self.positional2)
        ..add('named1', _self.named1))
      .toString();

  Test copyWith({
    int? positional1,
    String? positional2,
    bool? named1,
  }) {
    return Test(
      positional1: positional1 ?? _self.positional1,
      positional2: positional2 ?? _self.positional2,
      named1: named1 ?? _self.named1,
    );
  }

  Test change(void Function(_TestChanges c) updates) =>
      (_TestChanges._(_self)..update(updates)).build();

  _TestChanges toChanges() => _TestChanges._(_self);
}

class _TestChanges {
  int positional1;
  String positional2;
  bool named1;

  _TestChanges._(Test dc)
      : positional1 = dc.positional1,
        positional2 = dc.positional2,
        named1 = dc.named1;

  void update(void Function(_TestChanges c) updates) => updates(this);

  Test build() => Test(
        positional1: positional1,
        positional2: positional2,
        named1: named1,
      );
}
```

which contains  a `copyWith` method and a `_TestChanges` class that assume all parameters are named.

I have a simple use case where the first 2 parameters are obvious and the rest of the parameters are named and optional, and I don't wish to make the first 2 parameters named.

**Comment by BreX900** [comment]
@ykrasik The feature was released in version `1.1.0` of `mek_data_class_generator`.
I added code_builder for better readability of the code so there will be many changes in the generated code but everything will work as before

Thanks a lot for your support

**Comment by ykrasik** [comment]
Thanks, it works!
